server/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -1064,8 +1064,8 @@ private void registerFetchSubPhase(FetchSubPhase subPhase) {
     }
 
     private void registerRetrieverParsers(List<SearchPlugin> plugins) {
-        registerRetriever(new RetrieverSpec<>(StandardRetrieverBuilder.NAME, StandardRetrieverBuilder::fromXContent));
-        registerRetriever(new RetrieverSpec<>(KnnRetrieverBuilder.NAME, KnnRetrieverBuilder::fromXContent));
+        registerRetriever(new RetrieverSpec<>(StandardRetrieverBuilder.TYPE, StandardRetrieverBuilder::fromXContent));
+        registerRetriever(new RetrieverSpec<>(KnnRetrieverBuilder.TYPE, KnnRetrieverBuilder::fromXContent));
```

Comment:
Honestly, `NAME` is preferred. Why did you add all this churn to switch to `TYPE`? These are `namedWriteables` and their serialization is nominally based.

Reply:
I see that these aren't writeables, but still, parsing via xcontent is called ` p.namedObject`. I don't see a reason for this switch.

Reply:
This was changed in https://github.com/elastic/elasticsearch/commit/dada6f6d22a3ccd40bf47d02ca473e0fbcd0016a to avoid confusion, as I introduced a new `name` field for supporting `_name` param natively in retrievers (as a first attempt).

Reply:
@pmpailis retrieverName is fine + Javadocs is fine. I don't think we should add extra churn switching from `NAME` to `TYPE` to indicate the named serializable information